src/components/search-panel/SearchPanelForm.tsx

```diff
@@ -0,0 +1,134 @@
+/** ****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+import React, { useMemo, useState } from 'react';
+import {
+	DateTimeInputType,
+	DateTimeMask,
+	FilterRowConfig,
+	TimeInputType,
+} from '../../models/filter/FilterInputs';
+import { SearchPanelState } from './SearchPanel';
+import FilterRow from '../filter/row';
+import FilterDatetimeInput from '../filter/date-time-inputs/DateTimeInput';
+import { DATE_TIME_INPUT_MASK } from '../../util/filterInputs';
+
+interface SearchPanelFormProps {
+	formType: 'event' | 'message';
+	state: SearchPanelState;
+	setState: (
+		patch: Partial<SearchPanelState> | ((prevState: SearchPanelState) => Partial<SearchPanelState>),
+	) => void;
+}
+
+type DateInputProps = {
+	inputConfig: DateTimeInputType;
+};
+
+const SearchPanelForm = (props: SearchPanelFormProps) => {
+	const { formType, state, setState } = props;
+	const [currentStream, setCurrentStream] = useState('');
+
+	const toggleSearchDirection = () => {
+		setState(currentState => ({
+			...currentState,
```

Comment:
same. i thought you used useSetState hook to be able to set state in patches but you stil spread over previous state as you would do with useState hook